cli/silence_query.go

```diff
@@ -89,6 +90,11 @@ func query(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	within, err := queryFlags.GetDuration("within")
```

Comment:
For the sake of consistency, would it be possible to get the flag value as a string, and then parse it with the prometheus duration parser?

https://github.com/prometheus/alertmanager/commit/ea9a584e

It seems unlikely folks are interested in filtering down to silences that are expiring in weeks or even years, but we should offer the same duration parsing functionality.